reflex/state.py

```diff
@@ -1110,3 +1127,104 @@ def _convert_mutable_datatypes(
         )
 
     return field_value
+
+
+class ClientStorageBase:
+    """Base class for client-side storage."""
+
+    def options(self) -> dict[str, Any]:
+        """Get the options for the storage.
+
+        Returns:
+            All set options for the storage (not None).
+        """
+        return {
+            format.to_camel_case(k): v for k, v in vars(self).items() if v is not None
+        }
+
+
+class Cookie(ClientStorageBase, str):
+    """Represents a state Var that is stored as a cookie in the browser."""
+
+    name: str | None
+    path: str
+    max_age: int | None
+    domain: str | None
+    secure: bool | None
+    same_site: str
+
+    def __new__(
```

Comment:
Why do we use `__new__` instead of `__init__` ?

Reply:
these two classes extend `str`, which is immutable, and thus cannot override `__init__` to set values.